exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlWorker.java

```diff
@@ -226,7 +226,10 @@ private static SqlNode checkAndApplyAutoLimit(SqlConverter parser, QueryContext
     if (isAutoLimitShouldBeApplied(context, sqlNode)) {
       sqlNode = wrapWithAutoLimit(sqlNode, context);
     } else {
-      context.getOptions().setLocalOption(ExecConstants.QUERY_MAX_ROWS, 0);
+        //Force setting to zero IFF autoLimit was intended to be set originally but is inapplicable
```

Comment:
Please fix indentation and please refactor methods added in the previous commit to pass the value of `context.getOptions().getOption(ExecConstants.QUERY_MAX_ROWS).num_val.intValue()` instead of `context`.

Reply:
👍